src/main/java/uk/nhs/tis/trainee/notifications/migration/ResendAugust2025RotationScheduleFailures.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.quartz.JobDataMap;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import uk.nhs.tis.trainee.notifications.model.History;
+import uk.nhs.tis.trainee.notifications.service.HistoryService;
+import uk.nhs.tis.trainee.notifications.service.NotificationService;
+
+/**
+ * Resend email notifications which failed and got stuck in SCHEDULED.
+ */
+@Slf4j
+@ChangeUnit(id = "resendAugust2025RotationScheduleFailures", order = "10")
+public class ResendAugust2025RotationScheduleFailures {
+
+  private static final long WINDOW = Duration.ofDays(1).getSeconds();
+
+  private final MongoTemplate mongoTemplate;
+  private final HistoryService historyService;
+  private final NotificationService notificationService;
+
+  /**
+   * Construct the migrator.
+   *
+   * @param mongoTemplate The mongo template to use for accessing failed records.
+   */
+  public ResendAugust2025RotationScheduleFailures(MongoTemplate mongoTemplate,
+      HistoryService historyService, NotificationService notificationService) {
+    this.mongoTemplate = mongoTemplate;
+    this.historyService = historyService;
+    this.notificationService = notificationService;
+  }
+
+  /**
+   * Resend the failed Google email notifications.
+   */
+  @Execution
+  public void migrate() {
+    Query query = new Query()
+        .addCriteria(Criteria.where("recipient.type").is(EMAIL))
+        .addCriteria(Criteria.where("status").is(SCHEDULED))
+        .addCriteria(Criteria.where("sentAt")
+            .gte(LocalDate.of(2025, 5, 1))
+            .lt(LocalDate.now(ZoneOffset.UTC))
+        );
+
+    List<History> missedSchedules = mongoTemplate.find(query, History.class);
+    int total = missedSchedules.size();
+    log.info("Found {} qualifying missed schedules.", total);
+    int current = 1;
+
+    for (History missedSchedule : missedSchedules) {
+      log.info("Progress: {}/{}.", current++, total);
+      boolean success = scheduleEmail(missedSchedule);
+
+      if (success) {
+        historyService.deleteHistoryForTrainee(missedSchedule.id(),
+            missedSchedule.recipient().id());
+      }
+    }
+  }
+
+  /**
+   * Schedule retrying to send the missed email.
+   *
+   * @param missedSchedule The missed schedule notification history.
+   * @return Whether the notification was successfully scheduled.
+   */
+  private boolean scheduleEmail(History missedSchedule) {
+    Map<String, Object> templateVariables = missedSchedule.template().variables();
+    JobDataMap jobData = new JobDataMap(templateVariables);
+    String jobId = "%s-%s".formatted(missedSchedule.type(), missedSchedule.tisReference().id());
+
+    try {
+      notificationService.removeNotification(jobId);
+      notificationService.scheduleNotification(jobId, jobData, Date.from(Instant.now()), WINDOW);
```

Comment:
So we are still spreading these out over a day, despite the queue? I guess that's ok.